independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -803,11 +810,23 @@ private void ensureOpen() {
         }
     }
 
+    public CuratedApplication getCuratedApplication() {
+        return curatedApplication;
+    }
+
     @Override
     public String toString() {
         return "QuarkusClassLoader:" + name + "@" + Integer.toHexString(hashCode());
     }
 
+    public StartupAction getStartupAction() {
+        return startupAction;
+    }
+
+    public void setStartupAction(StartupAction startupAction) {
```

Comment:
This coupling looks like something we should review in a follow up.

Reply:
See comments in https://github.com/quarkusio/quarkus/pull/34681#discussion_r2005233801. I can't see a way of getting rid of it that doesn't involve global state ... but we can look at it, certainly.